### PR TITLE
2.6 - Guard SetApplicationConfig Against Empty Branch Name

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This patch adds logic already present for un-setting configuration values. 

It ensures that when an empty string is supplied for branch/generation name to `SetApplicationConfig`, it is defaulted to "master".

## QA steps

- Build and bootstrap this branch.
- `juju deploy redis`.
- Build or Snap install a 2.5.7 client.
- Use this client to `juju config redis password=pass`.
- Check that the password is set without error. 

## Documentation changes

None.

## Bug reference

https://bugs.launchpad.net/juju/+bug/1829140

### CODE REVIEW

**Comment by manadart** [comment]
[Branch Error Setting Charm Configuration](https://trello.com/c/UNMpdVQn/1395-branch-error-setting-charm-configuration)

File: apiserver/facades/client/application/application.go [hunk 1]
```diff
@@ -2146,6 +2146,12 @@ func (api *APIBase) setApplicationConfig(arg params.ApplicationConfigSet) error
 		if err != nil {
 			return errors.Trace(err)
 		}
+
+		// We need a guard on the API server-side for direct API callers such as
+		// python-libjuju. Always default to the master branch.
```
Comment (jameinel) [inline apiserver/facades/client/application/application.go]: it is also older Juju clients (2.5) not just libjuju

Comment (manadart) [reply apiserver/facades/client/application/application.go]: Added that detail.

**Comment by manadart** [comment]
I think this is a good spot for the 2.6 branch, as the generations/branches logic will probably remain stillborn there. Time will tell as far as new functionality goes.

**Comment by manadart** [comment]
$$merge$$

**Pull request merged**